### PR TITLE
Downloading files in parallel from s3

### PR DESCRIPTION
**Pull request by dav009**
This is a dummy attempt to speed a query I had (1GB, ~100 files)
 - master version: 27min
 - this branch: 6min

Probably not the best as this could be a pool of coroutines and avoid that global.

---
This change is [Reviewable](https://reviewable.io/reviews/zaiusinc/python-zaius-export/7)

### CODE REVIEW

**Review by netguy204 — COMMENTED** [review]
> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/zaiusinc/python-zaius-export/7)* status: all files reviewed, 1 unresolved discussion (waiting on @dav009)

___
*[zaius/export/api.py, line 148 at r1](https://reviewable.io/reviews/zaiusinc/python-zaius-export/7#-Mi2Uylr-mTZn2FpG9yb:-Mi2Uylv8q6QG3JsBrL0:b82tt3g) ([raw file](https://github.com/zaiusinc/python-zaius-export/blob/55740a433a84c50789467501354599a3ead4fb25/zaius/export/api.py#L148)):*
> ```Python
>         while True:
>             objs = s3_client.list_objects_v2(**kwargs)
>             keys = keys + [obj["Key"] for obj in objs["Contents"]]
> ```

use keys.extend instead to avoid unnecessary allocation
```
keys.extend([obj['Key'] for obj in objs['Contents']])
```

**Review by netguy204 — COMMENTED** [review]
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/zaiusinc/python-zaius-export/7)* status: all files reviewed, 1 unresolved discussion (waiting on @dav009)

File: zaius/export/api.py [hunk 1]
```diff
@@ -140,20 +142,46 @@ def _s3_download(self, s3_url, local_path):
...
             objs = s3_client.list_objects_v2(**kwargs)
-
-            for obj in objs["Contents"]:
-                _, fname = os.path.split(obj["Key"])
-                self.log.info("downloading {}".format(obj["Key"]))
-                s3_client.download_file(
-                    bucket, obj["Key"], os.path.join(local_path, fname)
-                )
-
+            keys.extend([obj["Key"] for obj in objs["Contents"]])
```
Comment (mal) [inline zaius/export/api.py:148]:
```suggestion
            keys.extend(obj["Key"] for obj in objs["Contents"])
```
Further to the use of `extend`, you can also use the comprehension directly as the argument here as it's iterable, which then avoids materialising it into a new list before passing it.

**Review by netguy204 — COMMENTED** [review]
:lgtm: 

> *[Reviewable](https://reviewable.io/reviews/zaiusinc/python-zaius-export/7#-:-MiRS3UDBZ3ap_kdxe35:bg7igkv)* status: all files reviewed, 1 unresolved discussion (waiting on @dav009)

**Review by gphillips8frw — APPROVED** [review]

**Pull request merged**